frontend/src/api/StreamApi.js

```diff
@@ -1,45 +1,31 @@
 import axios from "axios";
 import { convertToDataObject } from "../model/ConvertDataFormat";
 import StreamInfo from "../model/StreamInfo";
-import * as SampleData from "./SampleData";
 
-export function getStream(streamId) {
-  return new Promise((resolve, reject) => {
-    axios
-      .get(`${process.env.REACT_APP_STREAM}/${streamId}`)
-      .then((response) => {
-        const stream = response.data;
-        resolve(
-          new StreamInfo(
-            stream.id,
-            convertToDataObject(stream.outputChannel.encoder),
-            convertToDataObject(stream.inputChannel.decoder),
-            ["Additional stream info goes here."]
-          )
-        );
-      })
-      .catch(reject);
-  });
+export async function getStream(streamId) {
+  return axios
+    .get(`${process.env.REACT_APP_STREAM}/${streamId}`)
+    .then((response) => {
+      const stream = response.data;
+      return new StreamInfo(
+        stream.id,
+        convertToDataObject(stream.outputChannel.encoder),
+        convertToDataObject(stream.inputChannel.decoder),
+        ["Additional stream info goes here."]
+      );
+    });
 }
 
-export function getAllStreams(callback) {
-  axios
-    .get(process.env.REACT_APP_STREAM)
-    .then((streams) => {
-      Promise.all(
-        streams.data.map((streamId) => {
-          return getStream(streamId);
-        })
-      ).then(callback);
-    })
-    .catch(() => {
-      SampleData.getAllStreams(callback);
-    });
+export async function getAllStreams() {
+  return axios.get(process.env.REACT_APP_STREAM).then((streams) => {
+    return Promise.all(
+      streams.data.map((streamId) => {
+        return getStream(streamId);
+      })
+    );
```

Comment:
I think you could get away with creating an array of Streams then returning that instead of the Promise.all.

Reply:
Since getStream returns a promise (as it calls the backend) I don't really see a way to write this code without using `Promise.all`. This implementation waits for all the backend calls to complete and then creates an array from the returned streams. Could you maybe show me the code of the alternate implementation you suggest?

Reply:
On second thought I don't think it'll work, I didn't realize getStream returned a promise.